packages/flutter_test/lib/src/finders.dart

```diff
@@ -654,25 +1145,28 @@ class _HitTestableFinder extends ChainedFinder {
   }
 }
 
-/// Searches a widget tree and returns nodes that match a particular
-/// pattern.
-abstract class MatchFinder extends Finder {
-  /// Initializes a predicate-based Finder. Used by subclasses to initialize the
-  /// [skipOffstage] property.
-  MatchFinder({ super.skipOffstage });
-
+/// A mixin for creating finders that search candidates for those that match
+/// a given pattern.
+mixin MatchFinderMixin<CandidateType> on FinderBase<CandidateType> {
   /// Returns true if the given element matches the pattern.
   ///
   /// When implementing your own MatchFinder, this is the main method to override.
-  bool matches(Element candidate);
+  bool matches(CandidateType candidate);
 
   @override
-  Iterable<Element> apply(Iterable<Element> candidates) {
+  Iterable<CandidateType> findInCandidates(Iterable<CandidateType> candidates) {
     return candidates.where(matches);
   }
 }
 
-abstract class _MatchTextFinder extends MatchFinder {
+/// Searches candidates for any that match a particular pattern.
```

Comment:
mention that this is for widgets?

Reply:
It is not immediately clear to me why these intermediate classes (like MatchWidgetFinder and ChainedWidgetFinder) need to exist. They don't implement any logic and it is just another level of abstraction one needs to navigate when understanding how things work. Why can the finders that need it not just apply the mixin directly?

Reply:
> I'd probably un-rename them and get rid of the typedefs though. No reason to typedef things for backwards compatibility if we're just going to get rid of them eventually anyways.

That sounds good to me. Reducing the number of concepts one has to keep track would be a good idea, I think.

Reply:
Done. Well, prepped to be done with #127136 at least.